### PR TITLE
Add live reloading

### LINKED ISSUE #1166: Enable hot reloading with Gulp
**Describe the bug**
As a developer I expect the Electron app to live reload when I make changes to the CSS and JS files.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run `npm start`
2. Change any of the JS of CSS files.
3. The Electron app doesn't automatically reload.

**Expected behaviour**
The Electron app should reload when I make changes to a file so I don't have to keep restarting the app with `npm start`.

### PR DESCRIPTION
**Pull request by BronsonQuick**
Fixes #1166.

This includes https://github.com/igdmapps/igdm/pull/1170 so it should be merged first.

### CODE REVIEW

File: main/main.js [hunk 1]
```diff
@@ -45,6 +46,8 @@ function createWindow () {
     }))
   })
 
+  client.create(mainWindow);
```
Comment (ifedapoolarewaju) [inline main/main.js]: I think we should connect this client only when the app is in dev mode using [this condition check](https://github.com/igdmapps/igdm/pull/1171/files#diff-3617d9e96ff6d93de86492f30632549aL144)

Comment (BronsonQuick) [reply main/main.js]: Good thinking! I've added a conditional check now for that.

Comment (ifedapoolarewaju) [reply main/main.js]: hmm, the changes don't seem to appear here yet.

**Comment by BronsonQuick** [comment]
@ifedapoolarewaju Ahh sorry man. I forgot to push them! 🤦‍♂️ I've pushed them now!

File: main/main.js [hunk 2]
```diff
@@ -44,7 +45,10 @@ function createWindow () {
       slashes: true
     }))
   })
-
+  // If we're in development mode then create an Electron Connect client for live reload.
+  if (!process.defaultApp) {
```
Comment (ifedapoolarewaju) [inline main/main.js]: the defaultApp runs when in dev mode, so the condition should be:
```suggestion
  if (process.defaultApp) {
```

**Comment by ifedapoolarewaju** [comment]
Thank you!

**Pull request merged**